tpu-provisioner/internal/controller/deletion_controller.go

```diff
@@ -183,12 +184,15 @@ func (r *DeletionReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		return fmt.Errorf("NodeCriteria.MinLifetime must be set")
 	}
 
-	// NOTE: Direct Node watches are filtered based on labels in main.go.
-	//       However, Reconcile() can still be called for non-filtered Nodes
-	//       Because the of Watch on Pods below.
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&corev1.Node{}).
-		Watches(&source.Kind{Type: &corev1.Pod{}}, handler.EnqueueRequestsFromMapFunc(handler.MapFunc(nodeForPod))).
```

Comment:
This might cause issues: deletions might take a long time b/c reconcile might not be called when the Pods leave the Nodes.